From f718ac37e7b817faed3d733d706eb613f61215b9 Mon Sep 17 00:00:00 2001
Message-Id: <patch-v8-12.12-f718ac37e7b-20220118T155150Z-avarab@gmail.com>
In-Reply-To: <cover-v8-00.12-00000000000-20220118T155150Z-avarab@gmail.com>
References: <cover-v7-00.10-00000000000-20211228T004707Z-avarab@gmail.com>
	<cover-v8-00.12-00000000000-20220118T155150Z-avarab@gmail.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Wed, 3 Nov 2021 22:20:03 +0100
Subject: [PATCH v8 12/12] fixup for being on top of
 avar/drop-grep-prefix-length now

---
 diffcore-pickaxe.c | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/diffcore-pickaxe.c b/diffcore-pickaxe.c
index 32d194130e7..98d5f87f808 100644
--- a/diffcore-pickaxe.c
+++ b/diffcore-pickaxe.c
@@ -210,18 +210,17 @@ static void compile_pickaxe(struct diff_options *o)
 	    (!needle || !*needle))
 		BUG("should have needle under -G or -S");
 	if (opts & (DIFF_PICKAXE_REGEX | DIFF_PICKAXE_KIND_GS_MASK)) {
-		grep_init(&o->pickaxe_grep_opt, the_repository, NULL);
+		grep_init(&o->pickaxe_grep_opt, the_repository);
+		o->pickaxe_grep_opt.pattern_type_option = GREP_PATTERN_TYPE_ERE;
 #ifdef USE_LIBPCRE2
-		grep_commit_pattern_type(GREP_PATTERN_TYPE_PCRE, &o->pickaxe_grep_opt);
-#else
-		grep_commit_pattern_type(GREP_PATTERN_TYPE_ERE, &o->pickaxe_grep_opt);
+		o->pickaxe_grep_opt.pattern_type_option = GREP_PATTERN_TYPE_PCRE;
 #endif
 
 		if (o->pickaxe_opts & DIFF_PICKAXE_IGNORE_CASE)
 			o->pickaxe_grep_opt.ignore_case = 1;
 		if (opts & DIFF_PICKAXE_KIND_S &&
 		    !(opts & DIFF_PICKAXE_REGEX))
-			o->pickaxe_grep_opt.fixed = 1;
+			o->pickaxe_grep_opt.pattern_type_option = GREP_PATTERN_TYPE_FIXED;
 
 		append_grep_pattern(&o->pickaxe_grep_opt, needle, "diffcore-pickaxe", 0, GREP_PATTERN);
 		compile_grep_patterns(&o->pickaxe_grep_opt);
-- 
2.35.0.rc1.864.g57621b115b6

